iwyu_tool.py

```diff
@@ -151,6 +151,11 @@ def main(compilation_db_path, source_files, verbose, formatter, iwyu_args):
                 print('WARNING: \'%s\' not found in compilation database.' %
                       source)
 
+    # Remove entries with directory belonging to any ignore_dirs
+    if ignore_dirs:
+        entries[:] = [entry for entry in entries if all(not entry['directory']
```

Comment:
Ugh, nested comprehensions always makes my head spin. And the `all(not` feels like a double negation.

If you can default `ignore_dirs` to an empty list, you can introduce a nested helper function to make this more obvious:

    def is_ignored(dirpath):
         return any(dirpath.startswith(i) for i in ignore_dirs)

    entries[:] = [e for e in entries if not is_ignored(os.path.realpath(e['directory']))]

I think that would be easier to penetrate.